taskcluster/scripts/lint/update_buildconfig_from_gradle.py

```diff
@@ -18,51 +19,45 @@
+)
 
 
 def _get_upstream_deps_per_gradle_project(gradle_root, existing_build_config):
     project_dependencies = defaultdict(set)
     gradle_projects = _get_gradle_projects(gradle_root, existing_build_config)
 
-    for configuration in CONFIGURATIONS_WITH_DEPENDENCIES:
-        logger.info(
-            f"Looking for dependencies in {configuration} configuration "
-            f"in {gradle_root}"
-        )
-
-        # This is eventually going to fail if there's ever enough projects to make the
-        # command line too long. If that happens, we'll need to split this list up and
-        # run gradle more than once.
-        cmd = list(_DEFAULT_GRADLE_COMMAND)
-        for gradle_project in sorted(gradle_projects):
-            cmd.extend(
-                [f"{gradle_project}:dependencies", "--configuration", configuration]
-            )
-
-        # Parsing output like this is not ideal but bhearsum couldn't find a way
-        # to get the dependencies printed in a better format. If we could convince
-        # gradle to spit out JSON that would be much better.
-        # This is filed as https://bugzilla.mozilla.org/show_bug.cgi?id=1795152
-        current_project_name = None
-        for line in subprocess.check_output(
-            cmd, universal_newlines=True, cwd=gradle_root
-        ).splitlines():
-            # If we find the start of a new component section, update our tracking
-            # variable
-            if line.startswith("Project"):
-                current_project_name = line.split(":")[1].strip("'")
-
-            # If we find a new local dependency, add it.
-            if line.startswith("+--- project") or line.startswith(r"\--- project"):
-                project_dependencies[current_project_name].add(line.split(" ")[2])
+    logger.info(f"Looking for dependencies in {gradle_root}")
+
+    # This is eventually going to fail if there's ever enough projects to make the
+    # command line too long. If that happens, we'll need to split this list up and
+    # run gradle more than once.
+    cmd = list(_DEFAULT_GRADLE_COMMAND)
+    cmd.extend(
+        [f"{gradle_project}:dependencies" for gradle_project in sorted(gradle_projects)]
+    )
+
+    # Parsing output like this is not ideal but bhearsum couldn't find a way
+    # to get the dependencies printed in a better format. If we could convince
+    # gradle to spit out JSON that would be much better.
+    # This is filed as https://bugzilla.mozilla.org/show_bug.cgi?id=1795152
+    current_project_name = None
+    for line in subprocess.check_output(
+        cmd, universal_newlines=True, cwd=gradle_root
+    ).splitlines():
+        # If we find the start of a new component section, update our tracking
+        # variable
+        if line.startswith("Project"):
+            current_project_name = line.split(":")[1].strip("'")
+
+        # If we find a new local dependency, add it.
+        local_dep_match = _LOCAL_DEPENDENCY_PATTERN.search(line)
```

Comment:
`seach()` makes sure we search the full string and not just the beginning. Transitive dependencies are indented in the gradle output. E.g.:

```
releaseUnitTestRuntimeClasspath - Runtime classpath of compilation 'releaseUnitTest' (target  (androidJvm)).
[...]
+--- project :browser-state
|    +--- project :lib-state
|    |    +--- org.jetbrains.kotlin:kotlin-stdlib:1.7.20 (*)
[...]
|    |    +--- project :support-base (*)
|    |    \--- project :support-ktx
```